docs_nnx/guides/randomness.md

```diff
@@ -0,0 +1,245 @@
+
+`reseed` is a function that accepts an arbitrary graph node (this include pytrees of Flax NNX Modules) and some keyword arguments containing the new seed or key value for the `RngStream`s specified by the argument names. `reseed` will then traverse the graph and update the random state of the matching `RngStream`s, this includes both setting the `key` to a possibly new value and resetting the `count` to zero.
+
+Here's an example of how to using `reseed` to reset the random state of the `Dropout` layer and verify that the computation is identical to the first time the model was called:
+
+```{code-cell} ipython3
+model = Model(nnx.Rngs(params=0, dropout=1))
+x = jnp.ones((1, 20))
+
+y1 = model(x)
+y2 = model(x)
+
+nnx.reseed(model, dropout=1) # reset dropout RngState
+y3 = model(x)
+
+assert not jnp.allclose(y1, y2) # different
+assert jnp.allclose(y1, y3)     # same
+```
+
+## Splitting Rngs
+When interacting with transforms like `vmap` or `pmap` it is often necessary to split the random state such that each replica has its own unique state. This can be done in two way, either by manually splitting a key before passing it to one of the `Rngs` streams as shown below:
+
+```{code-cell} ipython3
+split_key = random.split(random.key(0), 5)
+rngs = nnx.Rngs(params=0, dropout=split_key)
+nnx.display(rngs)
+```
+
+Notice that both the `key` and the `count` for the `dropout` stream have a dimension of `5`. This strategy the benefit of being very explicit, but can be a bit limiting as it wouldn't allow the stream to be used outside of the transform:
+
+```{code-cell} ipython3
+try:
+  rngs.dropout()
+except ValueError as e:
+  print(e)
+```
+
+The other option is to use the `nnx.split_rngs` decorator which will automatically split the random state of any `RngStream`s found in the inputs of the function, and will automatically "lower" them once the function call ends so the `Rngs` can be used outside again. `split_rngs` allows passing Filters to the `only` keyword argument to select the `RngStream`s that should be split. Using `split_rngs` is useful in combination with a transform but here we will show a simple example without any transforms to illustrate the concept, we'll use `split_rngs` with a transform on the next section.
+
+```{code-cell} ipython3
+rngs = nnx.Rngs(params=0, dropout=1)
+
+@nnx.split_rngs(splits=5, only='dropout')
+def f(rngs: nnx.Rngs):
+  print('Inside:')
+  # rngs.dropout() # ValueError: fold_in accepts a single key...
+  nnx.display(rngs)
+
+f(rngs)
+
+print('Outside:')
+rngs.dropout() # works!
+nnx.display(rngs)
+```
+
+## Transforms
+As stated before, in Flax NNX random state is just another type of state, this means that there is nothing special about it regarding transforms. This means that you should be able to use the state handling APIs of each transform to get the results you want. In this section we will two examples of using random state in transforms, one with `pmap` and another one with `scan`.
```

Comment:
> one with `pmap` and another one with `scan`.

What about this to highlight the intention of both examples:
one with `pmap` to split multiple RNG keys and one with `scan` to broadcast a single RNG key.